sdlib/d/gc/global.d

```diff
@@ -46,14 +53,42 @@ private:
 		roots = (cast(const(void*)[]*) ptr)[0 .. length];
 
 		import d.gc.range;
-		roots[index] = makeRange(range);
+		if (range.length == 0) {
+			roots[index] = cast(void*[]) range;
+		} else {
+			roots[index] = makeRange(range);
+		}
+	}
+
+	void removeRootsImpl(const void* ptr) {
+		assert(mutex.isHeld(), "Mutex not held!");
+
+		// Search in reverse, since it's most likely for things to be removed
+		// in the reverse order they were added.
+		foreach_reverse (i; 0 .. roots.length) {
+			if (cast(void*) roots[i].ptr == ptr) {
```

Comment:
is is usually used int he codebase for pointer compare. Not a big deal.

Reply:
I'm not sure what this is saying. Just remove the comment?

Reply:
`is` as in the keyword, instead of `==` . Reading this again, this indeed looks like a confused sentence :P

Reply:
Oh wow, totally glossed over the double `is` lol.